src/Microsoft.VisualStudio.SlowCheetah.VS/SlowCheetahPackage.SolutionEvents.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.SlowCheetah.VS
+{
+    using Microsoft.VisualStudio.Shell;
+    using Microsoft.VisualStudio.Shell.Interop;
+
+    /// <summary>
+    /// Contains Solution Events
+    /// </summary>
+    public sealed partial class SlowCheetahPackage : Package, IVsUpdateSolutionEvents
```

Comment:
Make the solution events a nested class and have it implement `IDisposable`. The package can then construct it on initialize, and dispose it on dispose.